juju/juju
# network: add net-bond-reconfigure-delay model config option

This change introduces a model-config setting called net-bond-reconfigure-delay.

The value represents the amount of time in seconds to sleep between ifdown and ifup when bridging bonded interfaces; this is a platform bug and all of this can go away when bugs  #1594855 and #1269921 are fixed. Without a delay between ifdown and ifup bonded interfaces in 802.3ad LACP mode fail to come back up, rendering the network incomplete and therefore broken.

The default is set to 17 seconds. For a long time the bridge script hard-coded a value of 3 second but some setups now require an even longer duration. The last reported issue was fixed with a 10 second timeout, however, we're increasing that again because this issue (and solution) is not very discoverable and we want bridging to work out-of-the-box.

## QA steps

This setup requires your MAAS setup has nodes with bonded interfaces.

1. bootstrap on MAAS
2. Switch to the controller model
2. Verify the default value of net-bond-reconfigure-delay:
    $ juju model-config | grep net-bond-reconfigure-delay (should report 17)
3. Switch to the default model and change the value in this model
    $ juju model-config net-bond-reconfigure-delay=42
    $ juju model-config | grep net-bond-reconfigure-delay (should report 42)
4. Switch back to the controller and verify that it still has its own model config value of 17
5. In the default model add a container:
    $ juju add-machine lxd:0 --constraints spaces=space-0
6. On machine 0 run:
    $ sudo grep bridgescript /var/log/juju/machine-1.log
    You should observe that the bridge script has been invoked with --net-bond-reconfigure-delay=42
7. Change the value again .
    $ juju model-config net-bond-reconfigure-delay=67
8. Repeat steps (5) and (6); in (6) you should see your new period (67)

The documentation at: https://jujucharms.com/docs/2.0/models-config#list-of-model-keys needs updating to include the new key, its value and an explanation.

Fixes [LP:#1657579](https://bugs.launchpad.net/juju/+bug/1657579)
